.github/workflows/test_on_push.yaml

```diff
@@ -105,13 +105,13 @@ jobs:
         run: |
           nox -s examples
 
-  # Runs only on Ubuntu with Python 3.12
+  # Runs only on macos-14 with Python 3.12
   check_coverage:
     needs: style
-    runs-on: ubuntu-latest
+    runs-on: macos-14
```

Comment:
Question: is this being done because the macOS runners are known to be faster? I agree that this is quite beneficial, though!

Reply:
Yep, that's correct. The macOS arm64 runners are quite a bit fast than the x86 runners :)